distil/primitives/satellite_image_loader.py

```diff
@@ -128,7 +226,17 @@ def _load_image_group(self, uris, bands, base_uri: str) -> container.ndarray:
         for band, image in images:
             images_result[self._band_order[band.lower()]] = self._bilinear_resample(image, self._max_dimension)
 
+        # this is not optimized - should probably just be taking the image results above and appending those to a bytearray
```

Comment:
Can you make this change?  This is probably leads to some unnecessary memory churn.